apps/src/music/progress/MusicValidator.ts

```diff
@@ -110,6 +116,18 @@ export default class MusicValidator extends Validator {
 
         playedNumberSounds++;
 
+        // In order to check that the user has pressed the beat map buttons multiple times,
+        // we look at the unique invocation id. (Simple2 only)
+        if (eventData.triggered) {
+          const invocationId = eventData.functionContext?.uniqueInvocationId;
```

Comment:
Does this still work if you change the sounds under a trigger mid-playback? I think we re-sequence out the trigger events which means we'd have a new unique function ID.

That being said, maybe it's okay because we only want to count this if the same trigger with the same set of sounds is triggered more than once?

Reply:
The intention expressed by the curriculum team is just to be able to validate that a student pressed the same button repeatedly (to trigger the same sound). I didn’t manually test the scenario you describe, but I imagine they would need to click three times after making changes to their code.